cocotb/share/lib/simulator/simulatormodule.h

```diff
@@ -115,7 +115,12 @@ static PyMethodDef SimulatorMethods[] = {
         "--\n\n"
         "Set the log level for GPI"
     )},
-
+    {"is_running", is_running, METH_NOARGS, PyDoc_STR(
+        "is_running()\n"
+        "--\n\n"
+        "is_running() -> bool\n"
+        "Returns ``True`` if the caller is running within a simulator."
```

Comment:
Needs a newsfragment and maybe a `..versionadded`

Reply:
Done. Sorry for the late add. Forgot I wanted this in to address the needs of #1839.